### PR TITLE
[SPARK-42234][SQL] Rename error class: `UNSUPPORTED_FEATURE.REPEATED_PIVOT`

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to rename `UNSUPPORTED_FEATURE.REPEATED_PIVOT` to `REPEATED_CLAUSE`.

### Why are the changes needed?

`REPEATED_PIVOT` is actually not an `UNSUPPORTED_FEATURE`, and there must be other cases we should cover in more generic way

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Updated UTs.

### CODE REVIEW

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1207,6 +1207,12 @@
     ],
     "sqlState" : "42K03"
   },
+  "REPEATED_CLAUSE" : {
+    "message" : [
+      "The <clause> clause may be used at most once per <operation> operation."
+    ],
+    "sqlState" : "2202G"
```
Comment (srielau) [inline core/src/main/resources/error/error-classes.json:1214]: 2202G is: invalid repeat argument in a sample clause is quite specific.
```suggestion
    "sqlState" : "42616"
```

**Review by srielau — APPROVED** [review]

File: core/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (itholic) [reply core/src/main/resources/error/error-classes.json]: Just applied the suggestion!

Btw, seems like 42616 is missing in https://github.com/apache/spark/tree/master/core/src/main/resources/error.
Is it intentionally excluded or do we want to add it?

Comment (srielau) [reply core/src/main/resources/error/error-classes.json]: I mistypes. Should have been 42614 Duplicate clause.
42616 is Db2 LUW specific (I drew on DB2 zOS to fill the table

Comment (itholic) [reply core/src/main/resources/error/error-classes.json]: Thanks for the correction! Just updated to 42614.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master/3.4.
Thank you, @itholic and @srielau for review.

**Pull request closed without merging**